### DISCUSSION TITLE
how to read post params

### DISCUSSION THREAD

**Discussion by artknight** [uNetworking/uWebSockets.js#534] [Q&A]
Hi,
I am trying to get the post params ( content-type: 'x-www-form-urlencoded' ). 

In a regular node/express environment I would do something like
```
app.post('/*',(req, res) => {
  const user_id = req.body.id;
  const token = req.body.token;
  const geo = req.body.geo;
}
```

It does not seem to work here. Can anyone please point me in the right direction. Thank you!

**Comment by joshxyzhimself** [comment]
Don't you want content-type: application/json instead? There's an example in the /examples/ folder of this repo, and the client-side / browser-side code is easy with the fetch api. On top of it, with application/json content type you preserve types of string, number, boolean, and null; unlike urlencoded where everything is string.

**Reply by joshxyzhimself** [reply]
- https://googlechrome.github.io/samples/fetch-api/fetch-post.html
- https://github.com/uNetworking/uWebSockets.js/blob/master/examples/JsonPost.js

**Comment by sukiejosh** [comment]
```
app.post('/*',(req, res) => {
  const contentType = req.headers['content-type']
 //const user_id = req.body.id;
  //const token = req.body.token;
  //const geo = req.body.geo;
}
```
